File: Java/Q136SingleNumber.java
/*
@b-knd (jingru) on 24 July 2022 09:37:00
*/

import java.util.*;

/*
Concept: use XOR (^) to cancel out identical elements
*/

class Solution {
    public int singleNumber(int[] nums) {
        int ans = nums[0];
        for(int i = 1; i < nums.length; i++){
            ans ^= nums[i];
        }
        return ans;
    }
}

//Runtime: 1 ms, faster than 100.00% of Java online submissions for Single Number.
//Memory Usage: 50.9 MB, less than 35.91% of Java online submissions for Single Number.

File: Python/Q136SingleNumber.py
# @b-knd (jingru) on 09 August 2022 10:44:00

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = nums[0]
        for i, value in enumerate(nums[1:]):
            ans ^= value
        return ans
                
#Runtime: 172 ms, faster than 75.84% of Python3 online submissions for Single Number.
#Memory Usage: 16.7 MB, less than 50.74% of Python3 online submissions for Single Number.
